OpenRA.Mods.Common/Traits/Buildings/DockClientManager.cs

```diff
@@ -0,0 +1,297 @@
+#region Copyright & License Information
+/*
+ * Copyright (c) The OpenRA Developers and Contributors
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+using OpenRA.Mods.Common.Activities;
+using OpenRA.Mods.Common.Orders;
+using OpenRA.Primitives;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Traits
+{
+	public class DockClientManagerInfo : ConditionalTraitInfo
```

Comment:
I'd add a trait description that this is still very much tied to harvesters.

Reply:
The trait itself has no mention on any of Refinery / Harvester / FindAndDeliverResources. This whole abstraction is barely tied to harvester. IIRC only in 2 places, DockHost is tied to refinery as TSRefinery wasn't removed yet, and INotifyHarvesterAction interface hasn't had docking extracted